grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -350,10 +350,33 @@ class UserService {
     def getDashboardConfig(User user) {
         def config = grailsApplication.config.openboxes.tablero
         def userConfig = user.deserializeDashboardConfig()
-
+        Boolean configChanged = false
```

Comment:
@julienreliefapps Since config has a JSON format, would not it be easier to include in it, for each "section" a flag that indicates if it has changed in comparison to the one from Config.groovy? For example in Config.groovy by default:
`config.endpoints.graph.default = true` (or `changed = false`)
`config.endpoints.number.default = true` (or `changed = false`)
and each time if the user swaps something you just set this to the opposite and if the user restores the default value it will be set back to the former version. Thanks to that you don't have to do size counts or checking for key changes, only the check: `if not userConfig.number.default then do update`
cc @jmiranda

Reply:
Ok I think I understand, I ll try that @awalkowiak

Reply:
If I set the default to false everytime that the user update a new configuration and I just check this before resetting or not the configuration of the user, he will never keep it, everytime that he will refresh the page the configuration will be reset.

The problem is, yes when the user has a dashboard configuration saved, but only when the config.groovy has changed, so I don't understand how we can check if they are different just checking the user configuration

But maybe I have misunderstood what you proposed

Reply:
@julienreliefapps You are right this might not be the proper solution to the existing issue. But still, we need to find a better way of doing this.

Reply:
@awalkowiak  It's true, I ll try to find one